base_user_locale/controllers/web_client.py

```diff
@@ -0,0 +1,41 @@
+# Copyright 2019 Brainbean Apps (https://brainbeanapps.com)
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+import json
+
+from odoo.http import request, route
+
+from odoo.addons.web.controllers.main import WebClient, ensure_db
+
+
+class WebClient(WebClient):
+    def get_user_lang_parameters(self, user):
+        res = {}
+        date_format = user.date_format or user.company_id.date_format
```

Comment:
all these `user.company_id` should be `user.env.company`, isn't it?

Reply:
right